src/internal/middleware.js

```diff
@@ -91,5 +93,14 @@ export default function sagaMiddlewareFactory(options = {}) {
     return task
   }
 
+  sagaMiddleware.setContext = (props) => {
+    // TODO: or maybe `setContext` should be available on task and this one should act only as proxy method?
+    // would that be useful?
+    // const task = yield takeEvery(...)
+    // const task.setContext({...}) or maybe even - yield setContext(task, {...})?
```

Comment:
IMO the context of a task should be a private thing, just like `state` in React components. The component itself is an *owner* of its state and it shouldn't be accessible from its parent.